service/history/queue/processor_base.go

```diff
@@ -169,6 +171,11 @@ func (p *processorBase) updateAckLevel() (bool, error) {
 	totalPengingTasks := 0
 	for _, queueCollection := range p.processingQueueCollections {
 		ackLevel, numPendingTasks := queueCollection.UpdateAckLevels()
+		if ackLevel == nil {
```

Comment:
Looks like there are other places maybe need such check, can you double check and add if needed.
Another tiny improvement suggestion is on `queueCollection.UpdateAckLevels()`, check empty case at the beginning will make edge case easier to identify and save bits of ops when empty.

Reply:
Thanks for the suggestion. Added the same check when reseting queue states. Added the check for empty case in both `queueCollection.UpdateAckLevels()` and `queueCollection.Split()`.